### PR TITLE
Done command

### PR DESCRIPTION
**Pull request by kelvinnpc**
i changed the parser for delete a little for reuse in done

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/DoneCommand.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            }
+
+            taskDone = new Task(lastShownList.get(targetIndices[i] - 1));
+            taskDone.setStatus(new Status("done"));
+            tasksDoneList.add(taskDone);
+        	
+            try {
+                model.editTask(targetIndices[i], taskDone);
+            } catch (TaskNotFoundException pnfe) {
+                assert false : "The target task cannot be missing";
```
Comment (kabir-plod) [inline src/main/java/seedu/address/logic/commands/DoneCommand.java]: Shouldn't this return an IncorrectCommand instead? The user will re-enter his command with the feedback given.

File: src/main/java/seedu/address/logic/parser/Parser.java [hunk 2]
```diff
@@ -122,6 +127,36 @@ public Command parseCommand(String userInput) {
...
+			indices = prepareIndexList(arguments);
+		} catch (IncorrectCommandException e) {
+			return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, DoneCommand.MESSAGE_USAGE));
+		}
+
+		return new DoneCommand(indices);
+	}
+
+
+	private int[] prepareIndexList(String arguments) throws IncorrectCommandException{
```
Comment (kabir-plod) [inline src/main/java/seedu/address/logic/parser/Parser.java]: We might be able to combine prepareIndexList and parseIndices. But current implementation looks okay

File: src/main/java/seedu/address/logic/commands/DoneCommand.java (hunk 1 above)
Comment (kelvinnpc) [inline src/main/java/seedu/address/logic/commands/DoneCommand.java]: hmm, I following the delete command. I think this is not really an incorrect command, the parsing is correct just that the index given does not exist in the task manager

**Comment by kelvinnpc** [comment]
closes #17 #71

File: src/main/java/seedu/address/logic/commands/DoneCommand.java (hunk 1 above)
Comment (kabir-plod) [inline src/main/java/seedu/address/logic/commands/DoneCommand.java]: I think the assert silently fails the action without informing the user. The parsing is correct, yes, but the user should get a feedback message about the failure of his entry.

**Comment by kabir-plod** [comment]
Reviewed

**Comment by KnytJC** [comment]
reviewed

**Pull request merged**